Oanda/modules/trader/interpreter/interpreter.py

```diff
@@ -220,9 +228,20 @@ def trade(self, prediction, latest_value):
         filter_dict(data)
         
         print(readable_output(data))
+        
         try:
-            OrdersOrderCreate(self.access_token, self.accountID, data=data)
-            print("Bought ", units, " ", self.instrument, " value of trade: ", units*latest_value)
+            for i in range(tries):
+                if i < tries:
+                    raise MissingRepsonseException
+                if self.lastTransactionID == AccountDetails(self.access_token, self.accountID)[1]['lastTransactionID']:
+                    OrdersOrderCreate(self.access_token, self.accountID, data=data)
+                    time.sleep(wait)
```

Comment:
The failure mode that I saw before was this:
1. System attempts a trade by sending the trade message to Oanda
2. Something happens on the connection, and Oanda does not respond
3. Systems hangs (it waits for the response) for a long time, and the current job does not terminate

We need a timeout that stops the waiting after a number of seconds. What you added now does check if the trade indeed appears, but does it stop the system from waiting on a response?